src/validators/habit.validator.js

```
@@ -12,201 +13,146 @@ const isValidTimeZone = (value) => {
-    .withMessage('Timestamp must be a valid ISO 8601 date string.'),
-  body('timeZone')
-    .notEmpty()
-    .withMessage('TimeZone is required.')
-    .custom(isValidTimeZone)
-    .withMessage('Invalid IANA TimeZone format provided.'),
-  body('notes').optional({ nullable: true }).trim().isString(),
-];
+const habitIdParamSchema = z.object({
+  habitId: z
+    .string()
+    .refine((id) => !isNaN(parseInt(id)) && parseInt(id) > 0, {
```

ESLint

[error] Unexpected use of 'isNaN'. Use Number.isNaN instead https://github.com/airbnb/javascript#standard-library--isnan

(no-restricted-globals)